Add deployment info and set up Sentry releases

lemonsaurus opened the pull request:
Fixes #372 
Fixes #377

This adds two small changes related to git SHA.

## Add deployment info to all views
![image](https://user-images.githubusercontent.com/2098517/90943123-c5f90200-e418-11ea-919e-a3804783ae1b.png)

This is a useful bit of information that we can use to validate our assumptions about whether a deployment was successful.

## Set up Sentry Releases
This will allow us to pinpoint exactly which commit has introduced a new bug in Sentry, and should provide valuable context to errors we experience.

![image](https://user-images.githubusercontent.com/2098517/90944287-1b83dd80-e41e-11ea-8019-bbf5397a8c0f.png)

Commit:
Add gitpython ~=3.1.7 as a dependency.

We'll need this to fetch the hash.

Commit:
Install git in the docker environment.

This is required for gitpython to work.

Commit:
Add a context processor that outputs the git SHA.

This will make the SHA available in all templates.

Commit:
Add a comment with the git SHA to base templates.

Commit:
Optimize the git install in Dockerfile.

Commit:
Move git SHA initialization to __init__.py.

This will make it easier to use in multiple places.

Commit:
Set the sentry_sdk.init release to git_sha.

This is the first step in implementing releases for Sentry.

Commit:
Create a GitHub Action to make a Sentry release.

This should be all we need to enable the releases feature in Sentry for
this project.

Commit:
Oops, remove double git install from Dockerfile.

Commit:
Merge branch 'master' into add_deployment_info

Commit:
We need .git in the docker container for the SHA.

Commit:
Move git SHA fetcher into utils.

Commit:
Move git SHA fetcher into its own file.

Fix tests.

Commit:
omit resources.py from coverage reports.

I'm not writing the tests for that, it has nothing to do with this
branch.

MarkKoz commented on pydis_site/utils/utils.py:
```diff
@@ -0,0 +1,10 @@
+
+def get_git_sha() -> str:
+    """Get the Git SHA for this repo."""
+    return GIT_SHA
```
This seems redundant. Is this for coverage or something?

lemonsaurus replied:
Yep.

MarkKoz replied:
Ignoring the line with `# pragma: no cover` seems better than writing a function to get it to stop complaining. I don't know how to write a proper test for this anyway (like testing that the templates use it or something).

lemonsaurus replied:
sorry, I misread this part. you're saying why not just import the constant

Well, we could do that, I guess I mostly made this because it feels misleading to call it a util if it's not even a function, but that's not much of a justification.

MarkKoz replied:
How about calling it `constants.py` then?

lemonsaurus replied:
originally this function had all three lines of code, but then I realized it would be super wasteful to check the SHA every time this was called, so I moved it up to the top of the file. And now we're just left with, well, this.

MarkKoz commented on pydis_site/templates/wiki/base.html:
```diff
@@ -19,6 +19,7 @@
 {% endblock %}
 
 {% block content %}
+  <!-- Git hash for this release: {{ git_sha }} -->
```
Is this redundant? It causes the comment to show up twice for wiki pages

```html
<body class="site">
  <!-- Git hash for this release: 52d7ae050d32df7ceda6888f8a60590ceb0e37dc -->

<main class="site-content">
  

  
  <!-- Git hash for this release: 52d7ae050d32df7ceda6888f8a60590ceb0e37dc -->
  
    


<nav class="navbar is-primary" role="navigation" aria-label="main navigation">
```

lemonsaurus replied:
oh. my mistake, I was sure the wiki base didn't extend the other base. Removing it.

Commit:
Remove the SHA from the wiki base.html

This extends the regular base.html, so this would cause wiki pages to
have two SHA's.

Commit:
Move the SHA into constants.py.

The util was redundant. Thanks @MarkKoz

MarkKoz commented on pydis_site/constants.py:
```diff
@@ -0,0 +1,5 @@
+import git
+
+# Git SHA
+repo = git.Repo(search_parent_directories=True)
```
Maybe this should be "private" but I'm not too fussed at this point.

jb3 committed:
Merge branch 'master' into add_deployment_info
